js/reference.js

```diff
@@ -2797,6 +2850,35 @@
                 for (i = 0; i < compositeFKs.length; i++) {
                     this._referenceColumns.push(compositeFKs[i]);
                 }
+
+                // if array or true, add the remaining system columns
+                if (systemColumnsMode) {
+                    // array of column names to add
+                    var columnsToAdd = [];
+                    if (systemColumnsMode == true) {
+                        // add all, includes RID which will be skipped because of consideredColumns
+                        columnsToAdd = module._systemColumns;
+                    } else {
+                        // add the ones defined in array of config property
+                        // preserves order defined in `_systemColumns`
+                        columnsToAdd = module._systemColumns.filter(function (col) {
+                            return systemColumnsMode.indexOf(col) !== -1;
+                        });
+                    }
+
+                    for (i=0; i < columnsToAdd.length; i++) {
+                        var colName = columnsToAdd[i];
+                        if (colName in consideredColumns) {
+                            continue;
+                        }
+
+                        if (this._table.columns.has(colName)) {
```

Comment:
The `has` function will scan the array (it's calling filter). So it's just unnecessary to call it when you want to use the column object. you can just call the `get` function and then wrap it in try-catch.

```javascript
try {
  var column = this._table.columns.get(colName);
  addColumn(column);
} catch() {}
```

You should also do the `consideredColumns[colName] = true;` after the `if (colName in consideredColumns) `.